storage/innobase/handler/ha_innodb.cc

```diff
@@ -21877,6 +21906,180 @@ static void update_innodb_redo_log_encrypt(THD *thd, SYS_VAR *var,
   ut_ad(0);
 }
 
+/** Check if it is a valid value of innodb_undo_log_encryption_validate.
+Undo log encryption can be turned ON only if online to keyring
+encryption is disabled.
+@param[in]	thd	thread handle
+@param[in]	var	pointer to system variable
+@param[out]	save	immediate result for update function
+@param[in]	value	incoming bool
+@return 0 for valid innodb_track_changed_pages */
+static int innodb_undo_log_encryption_validate(THD *thd, SYS_VAR *var,
+                                               void *save,
+                                               struct st_mysql_value *value) {
+  const char *undo_log_encryption_input;
+  char buff[4];
+  int len = sizeof(buff);
+
+  ut_a(save != NULL);
+  ut_a(value != NULL);
+
+  undo_log_encryption_input = value->val_str(value, buff, &len);
+
+  if (len > 3) return 1;
+
+  bool is_undo_log_encryption_on =
+      !innobase_strcasecmp(undo_log_encryption_input, "ON") ||
+      !innobase_strcasecmp(undo_log_encryption_input, "1");
+
+  if (is_undo_log_encryption_on == false) {
+    if (innobase_strcasecmp(undo_log_encryption_input, "OFF") != 0 &&
+        innobase_strcasecmp(undo_log_encryption_input, "0") != 0) {
+      return 1;
+    }
+  } else if (Encryption::is_online_encryption_on()) {
+    push_warning_printf(thd, Sql_condition::SL_WARNING, ER_WRONG_ARGUMENTS,
+                        " Undo log cannot be"
+                        " encrypted with Master Key encryption"
+                        " when online to KEYRING encryption is turned ON.");
+    return 1;
+  } else {
+    undo::spaces->s_lock();
```

Comment:
Code shared with check_mk_and_keyring_encrypt_exclusion_for_undo, possible to refactor?

Reply:
done